win_scheduled_task module for windows

petemounce opened the pull request:
Fledgling module to allow scheduled tasks to be managed.

Note: At present, I only need enabled/disabled support. I'm using it to disable windows update scheduled tasks, since I'm managing that process myself. 

There's lots of scope for more features. See `Get-Help *Scheduled*` for inspiration :)

bcoca commented:
@cchurch

resmo commented on windows/win_scheduled_task.py:
```diff
@@ -0,0 +1,54 @@
+DOCUMENTATION = '''
+---
+module: win_scheduled_task
+version_added: "1.9"
```
update to "2.0"

resmo commented on windows/win_scheduled_task.py:
```diff
@@ -0,0 +1,54 @@
+      - Name of the scheduled task
+      - Supports * as wildcard
+    required: true
+    default: null
```
if required, there is no default. should be removed.

resmo commented on windows/win_scheduled_task.py:
```diff
@@ -0,0 +1,54 @@
+      - Supports * as wildcard
+    required: true
+    default: null
+    aliases: []
```
if there aren't aliases, no need to add a empty list. could be removed

petemounce commented:
@resmo thanks for the feedback; done.

gregdek commented:
Adding @owner_pr flag because @petemounce is co-owner of other Extras modules

bcoca commented:
instead of state=enabled|disabled, a enabled=yes|no flag is more within ansible norm. Also, could the module create/delete scheduled tasks, then state=present|absent will also be nice to have.

petemounce commented:
@bcoca I agree that adding create/delete would be a nice to have - but I don't need it, currently. It is a significant chunk of work that I won't have time to do in the short term.

perzizzle commented:
Has anyone signed up to add create/delete to this module? I have the need and could spend a little time working on it.

petemounce commented:
@perzizzle I have not, but happy to test those changes.

My particular use case for this module is disabling Windows update related tasks before baking an image, so I don't get punched with them running later and interesting with production traffic. Those tasks afaik can only be disabled, not deleted.

gregdek commented:
@petemounce is this ready for re-review in your opinion?

petemounce commented:
@gregdek yes. I have addressed all the points. I am not planning on increasing the scope to allow create/delete of scheduled tasks as @bcoca suggested - if that's required, then this should either not get merged, or get merged and an issue opened to come back and add it (@perzizzle was interested to implement).

perzizzle commented:
I'm still interested in adding this feature it just turned out to be non-trivial. I can have something ready within in the next week or two.

gregdek commented:
Thanks @perzizzle.

bcoca commented:
Ok with merging as is now and trying to add more features later on, incremental is good.

petemounce commented:
Thanks for the merge! :)
